ethereum/core/src/main/java/org/hyperledger/besu/ethereum/bonsai/worldview/BonsaiWorldState.java

```diff
@@ -282,7 +282,10 @@ private void updateAccountStorageState(
           worldStateUpdater.getAccountsToUpdate().get(updatedAddress);
       final BonsaiAccount accountOriginal = accountValue.getPrior();
       final Hash storageRoot =
-          (accountOriginal == null) ? Hash.EMPTY_TRIE_HASH : accountOriginal.getStorageRoot();
+          (accountOriginal == null
+                  || worldStateUpdater.getStorageToClear().contains(updatedAddress))
+              ? Hash.EMPTY_TRIE_HASH
+              : accountOriginal.getStorageRoot();
```

Comment:
@matthew1001 is this change related?

Reply:
Ummm - no - I'm confused as to where that's come in from. I looked at the files changed once I raised the PR and I don't think that was in there. But I can't explain how it did get there. I'll tidy up the branch in the morning to remove that commit (https://github.com/hyperledger/besu/pull/6285/commits/e90b4f1471de06935824db301f91c0a7222d3b1b) from the branch.

Reply:
That commit has now been removed